### ISSUE TITLE
aarch64 whl in PyPi

### ISSUE THREAD

**Issue by drikster80** [bitsandbytes-foundation/bitsandbytes#1437]
### Feature request

Requesting to distribute a Pypi whl for aarch64.

### Motivation

Nvidia's push to ARM64 architecture (e.g. GH200, NVL32/NV72) is driving a lot more projects to support this architecture. While downstream engines such as VLLM support aarch64, it would be more difficult to support official container distributions if they need to compile dependencies such as bitsandbytes from source for each build (https://github.com/vllm-project/vllm/pull/10499).

### Your contribution

I believe this would need to be done directly by the maintainers.

**Comment by johnnynunez** [comment]
@matthewdouglas nvidia is distributing wheels for aarch64 and sbsa:

https://pypi.jetson-ai-lab.dev/jp6/cu126
https://pypi.jetson-ai-lab.dev/jp6/cu128
https://pypi.jetson-ai-lab.dev/sbsa/cu128

how to install? ```pip3 install bitsandbytes --index-url https://pypi.jetson-ai-lab.dev/sbsa/cu128 ```

**Comment by matthewdouglas** [comment]
@johnnynunez Neat! Thanks for the heads up. At the moment I'm thinking we'll still want to build and push wheels for SBSA to PyPI here.

**Comment by drikster80** [comment]
⚠️⚠️⚠️ Call me suspicious, but that hosting domain (jetson-ai-lab.dev) isn't an official Nvidia repo (yet it links to an official nvidia URL (.com) on the main page, and the Who-is is private out of Iceland. The nameservers are using are from dynu.com (a DDNS service). At the time of this writing, the IP (108.39.248.12) has a GeoIP lookup of Pittsburg, PA in the united states and registered to Verizon.

It may work, but it's got enough red flags that I'm not going to touch those packages with a 10-foot pole right now.

@matthewdouglas, glad to hear you are still looking to build and push official packages!

**Comment by johnnynunez** [comment]
> ⚠️⚠️⚠️ Call me suspicious, but that hosting domain (jetson-ai-lab.dev) isn't an official Nvidia repo (yet it links to an official nvidia URL (.com) on the main page, and the Who-is is private out of Iceland. The nameservers it it using are from dynu.com (a DDNS service). At the time of this writing, the IP (108.39.248.12) has a GeoIP lookup of Pittsburg, PA in the united states and registered to Verizon FIOS.
> 
> It may work and be legit, but it's got enough red flags that I'm not going to touch those packages with a 10-foot pole right now.
> 
> [@matthewdouglas](https://github.com/matthewdouglas), glad to hear you are still looking to build and push official packages!

You are wrong.
https://www.jetson-ai-lab.com/

**Comment by dusty-nv** [comment]
Johnny jetson-ai-lab.dev is my house, its ok they flag it of course...  as the domain implies, it is for development.

That is good if BnB will build aarch64-sbsa cuda wheels and keep those current so pip "just works" for those users on Grace and DGX Spark.

We will continue building them for our source builds, and for Jetson Orin (which is aarch64 but not SBSA) but appreciate your support of ARM ecosystem since BnB is frequently installed with HF Transformers, and more recently I do use it as the performance has improved enough to be usable for those models that more optimized quantizations don't yet support.


________________________________

**Comment by drikster80** [comment]
Thanks for the heads up and confirmation. Personally, I don't have an issue with someone hosting on a local server/domain as long as it's attributable to an individual, but the original post said they were being distributed by Nvidia, so when it mapped to someone's home server it threw up major red flags for me.

Nvidia does have an official PyPi report (https://pypi.nvidia.com/). If it's possible to get those hosted on an Nvidia-owned domain, that would be fantastic and a huge help to the community.

Thanks for the work you guys are doing on this!
Also, I believe LambdaLabs is also hosting arm64-cuda wheels and the maintaining a vllm docker build:

https://github.com/LambdaLabsML/vllm-builder

**Comment by johnnynunez** [comment]
> Thanks for the heads up and confirmation. Personally, I don't have an issue with someone hosting on a local server/domain as long as it's attributable to an individual, but the original post said they were being distributed by Nvidia, so when it mapped to someone's home server it threw up major red flags for me.
> 
> Nvidia does have an official PyPi report (https://pypi.nvidia.com/). If it's possible to get those hosted on an Nvidia-owned domain, that would be fantastic and a huge help to the community.
> 
> Thanks for the work you guys are doing on this! Also, I believe LambdaLabs is also hosting arm64-cuda wheels and the maintaining a vllm docker build:
> 
> https://github.com/LambdaLabsML/vllm-builder

Yes… I did collaboration with them, so this is why sbsa is existing now

**Comment by matthewdouglas** [comment]
With #1539 and #1622 merged, we're building Linux aarch64 wheels now. We'll publish these to PyPI in the next release. In the meantime, the wheel built from main is available here:

https://github.com/bitsandbytes-foundation/bitsandbytes/releases/download/continuous-release_main/bitsandbytes-0.46.0.dev0-py3-none-manylinux_2_24_aarch64.whl
